src/components/application_manager/src/application_state.cc

```diff
@@ -204,8 +204,12 @@ void ApplicationState::RemoveHMIState(const WindowID window_id,
 
 void ApplicationState::EraseHMIState(HmiStates& hmi_states,
                                      HmiStates::iterator it) {
-  if (hmi_states.begin() == it) {
-    (*it)->set_parent(nullptr);
+  if (hmi_states.begin() == it && hmi_states.size() > 1) {
```

Comment:
```suggestion
  if (hmi_states.begin() == it) {
```
The checks within this statement should cover this. We don't want the `else` case triggering if `size() == 1` (though in practice this shouldn't happen)